### ISSUE TITLE
Add Qt support

### ISSUE THREAD

**Issue by dobokirisame** [dobokirisame/CompilerExplorer#13]
https://blog.qt.io/blog/2017/06/22/using-compiler-explorer-qt/

**Comment by dobokirisame** [comment]
- [ ] Add Qt includes like in article
- [ ] Find a way to get list of availabel Qt toolchains
- [ ] Add Qt includes from toolchains
- [ ] Create additional selector for Qt toolchain

**Comment by jcelerier** [comment]
hi, I just did a commit here : https://github.com/jcelerier/CompilerExplorer/commit/5653f3742f507416dc270ac2de0f02cbdf68f81c#diff-66f29047dffe593f0e23274d7ad2b6eeR182 that adds support for fetching automatically the correct include paths & defines that the Qt code model use (at least on linux with gcc :p). Maybe it can be useful for you ? (I can do a pull request but for now I have a few things that I don't really have time to clean up (I hardcoded the compilers because it wouldn't detect them correctly))

**Comment by dobokirisame** [comment]
Hello, @jcelerier. I've a look at your commit. Code for supporting -I and -D flags is very usefull, thank you. I'll add it to project asap. If you want you can make merge request and i'll accept it. But i can't accept the part where you return only local compilers. I did not hardcode this because i have 2 PC. First with linux on board and i build latest gcc and clang myself and place them in non-standart location. Second pc with win7 and MSYS2. It contains gcc and clang but in very specific place(somewhere in /c/Msys2/mingw64/usr/local/bin) That's why i created this task. I want to take compiler location form Qt toolchain.

**Comment by jcelerier** [comment]
> But i can't accept the part where you return only local compilers 

yep that's why I didn't do a PR ! but I couldn't get it to work with the compiler detection, maybe because the paths of my compiler have "/" in them (eg /usr/bin/g++) and they should be encoded in html chars ?
